File: backend/ResearchCruiseApp/Domain/Entities/Cruise.cs
using ResearchCruiseApp.Domain;
using ResearchCruiseApp.Domain;

namespace ResearchCruiseApp.Domain.Entities;

public class Cruise : Entity, IYearBasedNumbered
{
    public string Number { get; set; } = null!;

    public Guid MainCruiseManagerId { get; set; }

    public Guid MainDeputyManagerId { get; set; }
    public string StartDate { get; set; } = null!;
    public string EndDate { get; set; } = null!;

    public CruiseStatus Status { get; set; }
    public string? Title { get; set; }

    public bool ShipUnavailable { get; set; } = false;

    public List<CruiseApplication> CruiseApplications { get; set; } = null!;
}

File: frontend/src/types/Cruise.ts
import {CruiseApplication} from 'CruiseApplication';
import {CruiseApplicationShortInfo} from 'CruiseApplicationShortInfo';
import {CruiseStatus} from '@enums/CruiseStatus';

export type Cruise = {
    id: string;
    number: string;
    startDate: string;
    endDate: string;
    mainCruiseManagerId: string;
    mainCruiseManagerFirstName: string;
    mainCruiseManagerLastName: string;
    mainDeputyManagerId: string;
    cruiseApplicationsShortInfo: CruiseApplicationShortInfo[];
    cruiseApplications?: CruiseApplication[];
    status: CruiseStatus;
};
